backend/app/routers/session.py

```diff
@@ -19,8 +19,14 @@ async def create_session(
     db_session: Annotated[AsyncSession, Depends(get_session)],
     current_user: Annotated[User, Depends(get_current_user)],
     session_in: SessionIn,
+    request: Request,
 ) -> Session:
-    return await session_service.create_session(db_session, session_in.quiz_id, current_user)
+    return await session_service.create_session(
+        db_session,
+        session_in.quiz_id,
+        current_user,
+        f"{request.url.path}/{session_in.quiz_id}/join"
```

Comment:
Shouldn't the session ID in this case contain a session_id? If so, this will result in allowing for only one session per quiz or (because we don't have any constraint for it) introducing some ambiguity.

We don't know the session ID until we flush the db session so you need to adjust `create_session`. But still, the frontend should generate URLs IMO.